### PR TITLE
<Tomasz Ho Kei> whole-project PR for review

### PR DESCRIPTION
**Pull request by TomaszLaboj**

### CODE REVIEW

File: src/App.css
Comment (sgnahore) [inline src/App.css:1]: file name - convention is styles.css?

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+.header {
+  color: #f9f9f9;
+  text-align: center;
+}
+
+.searchText {
+  display: inline-block;
+}
+.search {
```
Comment (sgnahore) [inline src/App.css:18]: Inconsistency in spacing

File: src/App.tsx [hunk 2]
```diff
@@ -1,7 +1,74 @@
...
+    function fetchTVEpisodes() {
+      fetch("https://api.tvmaze.com/shows/83/episodes")
+        .then((res) => res.json())
+        .then((res) => setFullEpisodesArray(res));
+    }
+    fetchTVEpisodes();
+  }, []);
+
+  useEffect(() => {
+    const filteredArray: IEpisode[] = filterTVShowsBySearchInput(
```
Comment (sgnahore) [inline src/App.tsx:23]: Filtered array of what exactly?

File: src/App.tsx [hunk 3]
```diff
@@ -1,7 +1,74 @@
...
+      input,
+      fullEpisodesArray
+    );
+    setFilteredEpisodesArray(filteredArray);
+  }, [input, fullEpisodesArray]);
+  return (
```
Comment (sgnahore) [inline src/App.tsx:29]: Spacing

File: src/App.test.tsx
Comment (sgnahore) [inline src/App.test.tsx:1]: Could be best to remove this file rather than skip.

File: src/components/cleanUpSummary.test.ts
Comment (sgnahore) [inline src/components/cleanUpSummary.test.ts:1]: More test cases

File: src/components/formatEpisodeCode.test.ts
Comment (sgnahore) [inline src/components/formatEpisodeCode.test.ts:1]: More range in your test cases - edge cases in particular; what would happen in the case of 0 or 100 as inputs etc.

File: src/components/episodesComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { formatEpisodeCode } from "./formatEpisodeCode";
+import { cleanUpSummary } from "./cleanUpSummary";
+
+export interface IEpisode {
```
Comment (sgnahore) [inline src/components/episodesComponents.tsx:4]: Would be best to omit lines in the interface that are not read in the body of the file.

File: src/components/episodesComponents.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  _links: { self: { href: string } };
+}
+
+interface EpisodeViewProp {
+  episode: IEpisode;
+}
+
+export function DisplayEpisode(props: EpisodeViewProp): JSX.Element {
+  return (
+    <div className="episode">
```
Comment (sgnahore) [inline src/components/episodesComponents.tsx:30]: Maybe a clearer name here: i.e. episodeCard or episodeFrame etc.

File: src/components/filterTVShowsBySearchInput.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { IEpisode } from "./episodesComponents";
+
+export function filterTVShowsBySearchInput(
+  inputText: string,
+  inputArray: IEpisode[]
+): IEpisode[] {
```
Comment (sgnahore) [inline src/components/filterTVShowsBySearchInput.ts:6]: Spacing?

**Review by sgnahore — COMMENTED** [review]
Very well written and concise code, taken some good tips from the way some functions were written. Indentation was consistent and ideas have been communicated well. Only comment is some very minor inconsistencies.
